drivers/place/router/digraph.cr

```diff
@@ -0,0 +1,196 @@
+# Labelled digraph. Holds node labels of type *N* and edge labels of type *E*.
```

Comment:
This is neat. Are you planning on extracting this to a shard after a bit of battle testing?

Reply:
Sort of - this is an interface test. Medium term intent is to swap this out for https://git.sr.ht/~kb/sparse